pkg/resource/table/hooks_global_secondary_indexes.go

```diff
@@ -233,21 +233,21 @@ func (rm *resourceManager) newUpdateTableGlobalSecondaryIndexUpdatesPayload(
 
 // newSDKProvisionedThroughput builds a new *svcsdk.ProvisionedThroughput
 func newSDKProvisionedThroughput(pt *v1alpha1.ProvisionedThroughput) *svcsdk.ProvisionedThroughput {
-	provisionedThroughput := &svcsdk.ProvisionedThroughput{}
-	if pt != nil {
-		if pt.ReadCapacityUnits != nil {
-			provisionedThroughput.ReadCapacityUnits = aws.Int64(*pt.ReadCapacityUnits)
-		} else {
-			provisionedThroughput.ReadCapacityUnits = aws.Int64(0)
-		}
-		if pt.WriteCapacityUnits != nil {
-			provisionedThroughput.WriteCapacityUnits = aws.Int64(*pt.WriteCapacityUnits)
-		} else {
-			provisionedThroughput.WriteCapacityUnits = aws.Int64(0)
-		}
-	} else {
-		provisionedThroughput.ReadCapacityUnits = aws.Int64(0)
-		provisionedThroughput.WriteCapacityUnits = aws.Int64(0)
+	if pt == nil {
+		return nil
+	}
+	provisionedThroughput := &svcsdk.ProvisionedThroughput{
+		// ref: https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_ProvisionedThroughput.html
+		// Minimum capacity units is 1 when using provisioned capacity mode
+		ReadCapacityUnits:  aws.Int64(1),
+		WriteCapacityUnits: aws.Int64(1),
+	}
+	if pt.ReadCapacityUnits != nil {
+		provisionedThroughput.ReadCapacityUnits = aws.Int64(*pt.ReadCapacityUnits)
+	}
+
+	if pt.WriteCapacityUnits != nil {
+		provisionedThroughput.WriteCapacityUnits = aws.Int64(*pt.WriteCapacityUnits)
```

Comment:
Thoughts: What if we're not using PROVISIONED capacity mode? Would 0's be allowed? If that's the case would it be worth to evaluate the capacity mode in this function?

Reply:
I think you mean the `PAY_PER_REQUEST` mode. 
this is tricky part. IIRC,  we don't allow to set 0 capacity units for `PAY_PER_REQUEST` , please see the following message
```yaml
   conditions:
    - message: |
        InvalidParameter: 3 validation error(s) found.
        - minimum field size of 1, UpdateTableInput.GlobalSecondaryIndexUpdates[0].Create.Projection.NonKeyAttributes.
        - minimum field value of 1, UpdateTableInput.GlobalSecondaryIndexUpdates[0].Create.ProvisionedThroughput.ReadCapacityUnits.
        - minimum field value of 1, UpdateTableInput.GlobalSecondaryIndexUpdates[0].Create.ProvisionedThroughput.WriteCapacityUnits.
      status: "True"
      type: ACK.Terminal
```  

but aws DDB api will set capacity units as 0 when `PAY_PER_REQUEST`




> If you set BillingMode as PROVISIONED, you must specify this property. If you set BillingMode as PAY_PER_REQUEST, you cannot specify this property.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-provisionedthroughput

> If read/write capacity mode is PAY_PER_REQUEST the value is set to 0.

https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_ProvisionedThroughput.html